hudi-common/src/main/java/org/apache/hudi/common/table/timeline/CompletionTimeQueryView.java

```diff
@@ -175,42 +187,111 @@ public Option<String> getCompletionTime(String startTime) {
    *
    * <p>By default, assumes there is at most 1 day time of duration for an instant to accelerate the queries.
    *
-   * @param startCompletionTime The start completion time.
-   * @param endCompletionTime   The end completion time.
+   * @param readTimeline The read timeline.
+   * @param rangeStart   The query range start completion time.
+   * @param rangeEnd     The query range end completion time.
+   * @param rangeType    The range type.
    *
-   * @return The instant time set.
+   * @return The sorted instant time list.
    */
-  public Set<String> getStartTimeSet(String startCompletionTime, String endCompletionTime) {
+  public List<String> getStartTimes(
+      HoodieTimeline readTimeline,
+      Option<String> rangeStart,
+      Option<String> rangeEnd,
+      InstantRange.RangeType rangeType) {
     // assumes any instant/transaction lasts at most 1 day to optimize the query efficiency.
```

Comment:
How does this assumption come from? For large data replication or sql queries, it can be more than one day.

Reply:
That would be rare case, we can make it configurable if needed, but 1 day is generally enough for most of the use cases.